firmware/main.c

```diff
@@ -3,32 +3,128 @@
 PortData port_a;
 PortData port_b;
 
+// Indicates whether the SPI Daemon is listening for USB traffic
 volatile bool booted = false;
+// LED Chan: TCC1/WO[0]
+#define PWR_LED_TCC_CHAN 1
+// CC channel 0 on TCC instance 1
+# define PWR_LED_CC_CHAN 0
+// The maximum counter value was chosen to get a 2s period heartbeat
+#define MAX_COUNTER 0x16E30
+// We have 16 slices of a sine wave
+#define NUM_POINT_SLICES 0x10
+// Number of loop iterations in a single slice
+#define COUNTS_IN_SLICE MAX_COUNTER/NUM_POINT_SLICES
+// Evenly spaced points along a sine wave, shifted up by 1, scaled by 0.5
+// multiplied by the max counter value
+const float sin_wave_points[] = {
```

Comment:
Use `uint16_t` instead of float, so as to not pull in floating-point emulation libs.